app/controllers/admin_controller.rb

```diff
@@ -0,0 +1,8 @@
+# Controller to handle mediations for admins
+class AdminController < ApplicationController
+
+  def index
+    @requests = Request.all.group_by(&:origin).sort
+    authorize! :index, @requests
```

Comment:
Would something like `authorize! :manage, Request` work here?  If it does we can actually do the authorization before we hit the database (since we won't need the `@requests` variable).